strawberry/arguments.py

```diff
@@ -1,25 +1,110 @@
+from __future__ import annotations
+
 import enum
 import inspect
-from typing import Any, Dict, List, Mapping, Optional, Type, cast
+from typing import Any, Dict, List, Mapping, Optional, Type, Union, cast
 
 from typing_extensions import Annotated, get_args, get_origin
 
 from .exceptions import MultipleStrawberryArgumentsError, UnsupportedTypeError
 from .scalars import is_scalar
-from .types.types import ArgumentDefinition, undefined
+from .types.types import undefined
+from .union import StrawberryUnion
 from .utils.str_converters import to_camel_case
 
 
-class StrawberryArgument:
+class StrawberryArgumentAnnotation:
     description: Optional[str]
 
     def __init__(self, description: Optional[str] = None):
         self.description = description
 
 
+class StrawberryArgument:
+    def __init__(
+        self,
+        # TODO: this optional will probably go away when we have StrawberryList
+        python_name: Optional[str],
+        graphql_name: Optional[str],
+        type_: Optional[Union[Type, StrawberryUnion]],
+        origin: Optional[Type] = None,
+        child: Optional["StrawberryArgument"] = None,
+        is_subscription: bool = False,
+        is_optional: bool = False,
+        is_child_optional: bool = False,
+        is_list: bool = False,
+        is_union: bool = False,
+        description: Optional[str] = None,
+        default_value: Any = undefined,
+    ) -> None:
+        self.python_name = python_name
+        self._graphql_name = graphql_name
+        self.type = type_
+        self.origin = origin
+        self.child = child
+        self.is_subscription = is_subscription
+        self.is_optional = is_optional
+        self.is_child_optional = is_child_optional
+        self.is_list = is_list
+        self.is_union = is_union
+        self.description = description
+        self.default_value = default_value
+
+    @property
+    def graphql_name(self) -> Optional[str]:
+        if self._graphql_name:
+            return self._graphql_name
+        if self.python_name:
+            return to_camel_case(self.python_name)
+        return None
```

Comment:
Should add a test for coverage here

Reply:
I'd say let's skip it for now, as this will change when we have StrawberryList :)

Reply:
Why would `StrawberryList` change the name calculation?

Reply:
we only have python_name equal to None when using lists (children of lists have no python name nor graphql name)

Reply:
Oh right :slightly_smiling_face: